compiler/rustc_middle/src/ty/context.rs

```diff
@@ -409,6 +412,12 @@ impl<'tcx> Deref for TyCtxt<'tcx> {
     }
 }
 
+// `TyCtxt` may contain non-shared data structures only at initialization
+// time. We can determine the thread safety of `TyCtxt` by checking at
+// runtime whether these non-shared data structures actually exist.
+unsafe impl<'tcx> DynSendCheck for TyCtxt<'tcx> {}
+unsafe impl<'tcx> DynSyncCheck for TyCtxt<'tcx> {}
```

Comment:
These impls should be removed. The compiler is supposed to prove that `TyCtxt<'tcx>` is `DynSyncCheck` / `DynSendCheck`.

Reply:
Done